### PR TITLE
ORC-597: Add Row-level filtering benchmark

### PR DESCRIPTION
**Pull request by pgaref**
### What changes were proposed in this pull request?

Extending orc benchmark to support row-level filtering using existing datasets

### Why are the changes needed?

To compare the result between versions.

### How was this patch tested?

Run the benchmark.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @pgaref . Could you resolve the conflicts?

**Comment by pgaref** [comment]
> Hi, @pgaref . Could you resolve the conflicts?

Hey @dongjoon-hyun just rebased with the latest master -- it should solve the conflicts

File: java/bench/core/pom.xml [hunk 1]
```diff
@@ -112,6 +112,10 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
+          <configuration>
+              <source>8</source>
```
Comment (dongjoon-hyun) [inline java/bench/core/pom.xml]: indentation?

File: java/core/src/java/org/apache/orc/impl/IntegerReader.java [hunk 2]
```diff
@@ -67,6 +67,7 @@ void nextVector(ColumnVector column,
                    int length
                    ) throws IOException;
 
+
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/impl/IntegerReader.java]: Shall we revert this change?

**Comment by pgaref** [comment]
> Eventually, this PR should be split into two PRs: (1) reader change, and (2) add benchmark code.
> 
> > Extending orc benchmark to support row-level filtering using existing datasets

Hey @dongjoon-hyun -- The effort is already split in [ORC-577](https://issues.apache.org/jira/browse/ORC-577) adding the reader functionality and [ORC-597](https://issues.apache.org/jira/browse/ORC-597) extending the benchmarks  --  as I did not have permissions to create a branch upstream this PR also contains the Reader changes -- apologies for the mixup.

**Comment by pgaref** [comment]
Hey @dongjoon-hyun @jcamachor  could you please take a look?
Now that ORC-577 is merged this could be pretty useful.
Thanks!

**Comment by dongjoon-hyun** [comment]
cc @omalley
